charts/k8ssandra-cluster/templates/stargate/stargate.yaml

```diff
@@ -0,0 +1,75 @@
+{{- if .Values.stargate.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}-k8ssandra-stargate
+  labels:
+    app: {{ .Release.Name }}-k8ssandra-stargate
+{{ include "k8ssandra-cluster.labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.stargate.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Release.Name }}-k8ssandra-stargate
+  template:
+    metadata:
+      labels:
+        app: {{ .Release.Name }}-k8ssandra-stargate
+    spec:
+      containers:
+        - name: {{ .Release.Name }}-k8ssandra-stargate
+  {{- if eq .Values.stargate.containerImage "auto" }}
+    {{- if eq .Values.stargate.clusterVersion "3.11" }}
+          image: "stargateio/stargate-3_11:v0.0.9"
+    {{- else if eq .Values.stargate.clusterVersion "4.0" }}
+          image: "stargateio/stargate-4_0:v0.0.9"
+    {{- else }}
+          image: "containerImage-auto-did-not-recognize-clusterVersion-{{ .Values.stargate.clusterVersion }}"
+    {{- end }}
+  {{- else }}
+          image: {{ .Values.stargate.containerImage | quote }}
+  {{- end }}
+          ports:
+            - name: graphql
+              containerPort: 8080
+            - name: authorization
+              containerPort: 8081
+            - name: rest
+              containerPort: 8082
+            - name: admin
+              containerPort: 8084
+            - name: metrics
+              containerPort: 8085
+          resources:
+            limits:
+              memory: {{ .Values.stargate.resourceLimits.memory | quote }}
+          env:
+            - name: JAVA_OPTS
+              value: {{ .Values.stargate.javaOpts | quote }}
+            - name: CLUSTER_NAME
+              value: {{ .Values.clusterName | quote }}
+            - name: CLUSTER_VERSION
+              value: {{ .Values.stargate.clusterVersion | quote }}
+            - name: SEED
+              value: "{{ .Values.clusterName }}-seed-service"
+            - name: DATACENTER_NAME
+              value: {{ .Values.datacenterName | quote }}
+            - name: RACK_NAME
+              value: {{ .Values.rackName | quote }}
+            - name: ENABLE_AUTH
```

Comment:
It looks like we need auth enabled by default. I thought that enabling it would require Cassandra auth to be enabled. That does not appear to be the case. 

@dougwettlaufer even with auth disabled, Stargate is still telling me I need an auth token when I try using the REST API. Can you explain that?

The `AuthnTableBasedService` in Stargate queries the `system_auth.roles` table, but based on some initial testing, it looks like we do not have to have auth enabled in Cassandra. cass-operator adds a user in the `roles` table, so that may be sufficient. 

@dougwettlaufer can you confirm whether we need to have auth enabled in Cassandra? We have some work to do before we are ready to fully support auth in k8ssandra so hopefully the answer is no :)

Reply:
@jsanda as of now you need to have auth enabled for stargate to use the HTTP based services, even if it's just using the cassandra/cassandra superuser. It doesn't necessarily need to be enabled on the C* side though.